gtsam/hybrid/HybridGaussianConditional.cpp

```diff
@@ -28,56 +28,60 @@
 #include <gtsam/linear/GaussianFactorGraph.h>
 
 namespace gtsam {
-HybridGaussianFactor::FactorValuePairs GetFactorValuePairs(
-    const HybridGaussianConditional::Conditionals &conditionals) {
-  auto func = [](const GaussianConditional::shared_ptr &conditional)
-      -> GaussianFactorValuePair {
-    double value = 0.0;
-    // Check if conditional is pruned
-    if (conditional) {
-      // Assign log(\sqrt(|2πΣ|)) = -log(1 / sqrt(|2πΣ|))
-      value = conditional->negLogConstant();
-    }
-    return {std::dynamic_pointer_cast<GaussianFactor>(conditional), value};
-  };
-  return HybridGaussianFactor::FactorValuePairs(conditionals, func);
-}
+/* *******************************************************************************/
+struct HybridGaussianConditional::ConstructorHelper {
+  KeyVector frontals, parents;
+  HybridGaussianFactor::FactorValuePairs pairs;
+  double negLogConstant;
+  /// Compute all variables needed for the private constructor below.
+  ConstructorHelper(const Conditionals &conditionals) {
+    negLogConstant = std::numeric_limits<double>::infinity();
+
+    auto func = [&](const GaussianConditional::shared_ptr &c)
+        -> GaussianFactorValuePair {
+      double value = 0.0;
+      if (c) {
+        if (frontals.empty()) {
+          frontals = KeyVector(c->frontals().begin(), c->frontals().end());
+          parents = KeyVector(c->parents().begin(), c->parents().end());
+        }
+        value = c->negLogConstant();
+        negLogConstant = std::min(negLogConstant, value);
+      }
+      return {std::dynamic_pointer_cast<GaussianFactor>(c), value};
```

Comment:
Because we store the value, we should never have to call negLogConstant in this file again!
And that’s why we currently need to keep conditionals :-( at least until the hiding is removed.